core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchImpl.java

```diff
@@ -0,0 +1,134 @@
+package com.valtech.aem.saas.core.fulltextsearch;
+
+import static com.valtech.aem.saas.core.fulltextsearch.SearchImpl.RESOURCE_TYPE;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ContainerExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.day.cq.i18n.I18n;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.valtech.aem.saas.api.fulltextsearch.Filter;
+import com.valtech.aem.saas.api.fulltextsearch.Search;
+import com.valtech.aem.saas.api.fulltextsearch.SearchResults;
+import com.valtech.aem.saas.core.common.request.RequestParameters;
+import com.valtech.aem.saas.core.common.resource.ResourceChildren;
+import com.valtech.aem.saas.core.i18n.I18nProvider;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Default;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.ExporterOption;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ChildResource;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.apache.sling.models.factory.ModelFactory;
+
+@Model(adaptables = {SlingHttpServletRequest.class, Resource.class},
+    adapters = {Search.class, ComponentExporter.class, ContainerExporter.class},
+    defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL,
+    resourceType = RESOURCE_TYPE)
```

Comment:
It is needed for the model exporter framework. https://sling.apache.org/documentation/bundles/models.html#associating-a-model-class-with-a-resource-type-since-130-1